src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp

```diff
@@ -918,11 +918,47 @@ const TypeIdentifier TypeObjectRegistry::get_complementary_type_identifier(
 {
     if (TypeObjectUtils::is_direct_hash_type_identifier(type_id))
     {
-        std::lock_guard<std::mutex> data_guard(type_object_registry_mutex_);
+        std::unique_lock<std::mutex> lock(type_object_registry_mutex_);
         auto it = type_registry_entries_.find(type_id);
         if (type_registry_entries_.end() != it)
         {
-            return it->second.complementary_type_id;
+            if (TK_NONE != it->second.complementary_type_id._d())
+            {
+                return it->second.complementary_type_id;
+            }
+            else
+            {
+                TypeRegistryEntry minimal_entry;
```

Comment:
This is assuming that the input `type_id` is `EK_COMPLETE`

Reply:
Done